lib/geocoder/request.rb

```diff
@@ -75,6 +77,20 @@ def geocoder_split_ip_addresses(ip_addresses)
     def geocoder_reject_trusted_ip_addresses(ip_addresses)
       ip_addresses.reject { |ip| trusted_proxy?(ip) }
     end
+
+    def geocoder_remove_port_from_addresses(ip_addresses)
+      ip_addresses.map { |ip| ip.split(':').first }
+    end
+
+    def geocoder_reject_non_ipv4_addresses(ip_addresses)
+      reg = Regexp.new("^([01]?\\d\\d?|2[0-4]\\d|25[0-5])\\.([01]?\\d\\d?|2[0-4]\\d|25[0-5])\\.([01]?\\d\\d?|2[0-4]\\d|25[0-5])\\.([01]?\\d\\d?|2[0-4]\\d|25[0-5])$")
```

Comment:
Thanks for the tip. Yes. I wan't happy to see the long regex either. 

Using `IPAddr` is really simple and reliable. Great.